api/v1beta1/provider_types.go

```diff
@@ -22,7 +22,8 @@ import (
 )
 
 const (
-	ProviderKind string = "Provider"
+	ProviderKind     string = "Provider"
+	ProviderIndexKey string = ".metadata.provider"
```

Comment:
Not sure if this value belongs in the API package, as based on the implementation it seems to be part of the application logic, and would have little value to consumers.